rules/REQUEST-944-APPLICATION-ATTACK-JAVA.conf

```diff
@@ -246,6 +246,31 @@ SecRule ARGS|ARGS_NAMES|REQUEST_COOKIES|!REQUEST_COOKIES:/__utm/|REQUEST_COOKIES
     setvar:'tx.anomaly_score_pl2=+%{tx.critical_anomaly_score}'"
 
 
+# This rule is also triggered by the following exploit(s):
+# - https://www.rapid7.com/blog/post/2022/03/30/spring4shell-zero-day-vulnerability-in-spring-framework/
+# - https://www.ironcastle.net/possible-new-java-spring-framework-vulnerability-wed-mar-30th/
+#
+SecRule ARGS|ARGS_NAMES|REQUEST_COOKIES|!REQUEST_COOKIES:/__utm/|REQUEST_COOKIES_NAMES|REQUEST_BODY|REQUEST_HEADERS|XML:/*|XML://@* \
+    "@rx (?:class[.]module[.]classLoader[.]resources[.]context[.]parent[.]pipeline|springframework[.]context[.]support[.]FileSystemXmlApplicationContext)" \
```

Comment:
Why `[.]` instead of `\.`?

Reply:
oh, that's right! (I always have a problem with escape sequences importing regexs on other tools and I usually replace \. with [.]). I'm going to replace all [.] with \. thanks!

Reply:
done